casanovo/denovo/model_runner.py

```diff
@@ -54,11 +54,13 @@ def __init__(
         self.writer = None
 
         # Configure checkpoints.
-        if config.save_model:
+        if config.save_top_k is not None:
             self.callbacks = [
                 pl.callbacks.ModelCheckpoint(
                     dirpath=config.model_save_folder_path,
-                    save_top_k=-1,
+                    monitor="valid_CELoss",
```

Comment:
Currently the model doesn't save the weights at the first checkpoint (determined by `every_n_train_steps`) despite running validation and only does so at subsequent checkpoints. This warning is the only trace of something going on under the hood:

> 2023-05-10 12:47:47,393 WARNING [py.warnings/MainProcess] warnings._showwarnmsg : /net/noble/vol1/home/melih/anaconda3/envs/casanovo_dev_v2/lib/python3.8/site-packages/lightning/pytorch/utilities/data.py:76: UserWarning: Trying to infer the `batch_size` from an ambiguous collection. The batch size we found is 128. To avoid any miscalculations, use `self.log(..., batch_size=batch_size)`.
  warning_cache.warn(

I think this has to do with the order of validation with its logging and checkpointing (i.e. checkpointing is attempted before the results from 1st validation are logged) but I'm not entirely sure how to fix it. In the worst case, current checkpointing but might be lagging behind validation by 1 step - leading to a mismatch between validation results and checkpoints.

Reply:
I think I was able to fix this - @melihyilmaz can you try it out and see if you observe the correct behavior?